build/cmake/MaintainerFlags.cmake

```
@@ -17,7 +17,7 @@ if (ENABLE_MAINTAINER_FLAGS AND NOT MSVC AND NOT MONGOC_MAINTAINER_FLAGS_SET)
       set (CMAKE_REQUIRED_QUIET 0)
       if (FLAG_OK)
          message (STATUS "C compiler accepts ${MAINTAINER_FLAG}")
-         set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${MAINTAINER_FLAG}")
+         set (CMAKE_C_FLAGS "${MAINTAINER_FLAG} ${CMAKE_C_FLAGS}")
```

A nice alternative in CMake 3.10, but more self-documenting:

```suggestion
         string (PREPEND CMAKE_C_FLAGS "${MAINTAINER_FLAGS} ")
```

---

Alas the CMake binary used by the `build-and-test-with-toolchain` task (`/opt/mongo-c-toolchain/bin/cmake`) uses CMake 3.1.0 which does not support `string(PREPEND ...)`.